src/lbaf/IO/lbsMeshWriter.py

```diff
@@ -0,0 +1,306 @@
+            object_jitter: coefficient of random jitter with magnitude < 1
+            f: file name stem
+            r: grid_resolution value
+            output_dir: output directory
+        """
+        # Assign logger to instance variable
+        self.__logger = logger
+
+        # Ensure that provided phase has correct type
+        if not isinstance(p, Phase):
+            self.__logger.error("Could not write to ExodusII file by lack of a LBS phase")
+            sys.exit(1)
+        self.__phase = p
+
+        # Retrieve number of mesh points and bail out early if empty set
+        if not (n_p := len(self.__phase.get_ranks())):
+            self.__logger.error("Empty list of ranks, cannot write a mesh file")
+            return
+        self.__n_p = n_p
+
+        # Ensure that an iterable of grid sizes was passed
+        if not isinstance(grid_size, (list, tuple)):
+            self.__logger.error("Could not generate meshes by lack of a grid size iterable")
+            sys.exit(1)
+        self.__grid_size = grid_size
+        self.__logger.info(
+            f"Mapping {self.__n_p} ranks onto a {grid_size[0]}x"
+            f"{grid_size[1]}x{grid_size[2]} rectilinear grid")
+
+        # Ensure that a valid object jitter coefficient was passed
+        if not isinstance(object_jitter, float) or abs(object_jitter) >= 1.0:
+            self.__logger.error(f"Could not generate meshes due to unsuitable object jitter coefficient: {object_jitter}")
+            sys.exit(1)
+        self.__object_jitter = object_jitter
+        
+        # Ensure that specified grid resolution is correct
+        if not isinstance(r, numbers.Number) or r <= 0.:
+            self.__logger.error("Grid resolution must be a positive number")
+            sys.exit(1)
+        self.__grid_resolution = float(r)
+
+        # Assemble file and path names from constructor parameters
+        self.__rank_file_name = f"{f}_rank_view.e"
+        self.__object_file_name = f"{f}_object_view"
+        self.__output_dir = output_dir
+        if self.__output_dir is not None:
+            self.__rank_file_name = os.path.join(
+                self.__output_dir,
+                self.__rank_file_name)
+            self.__object_file_name = os.path.join(
+                self.__output_dir,
+                self.__object_file_name)
+
+
+    def global_id_to_cartesian(self, flat_id, grid_sizes):
+        """Map global index to its Cartesian grid coordinates."""
+
+        # Sanity check
+        n01 = grid_sizes[0] * grid_sizes[1]
+        if flat_id < 0 or flat_id >= n01 * grid_sizes[2]:
+            return None, None, None
+
+        # Compute successive Euclidean divisions
+        k, r = divmod(flat_id, n01)
+        j, i = divmod(r, grid_sizes[0])
+
+        # Return Cartesian coordinates
+        return i, j, k
+
+
+    def write_rank_view_file(self, distributions: dict, statistics: dict):
+        """ Map ranks to grid and write ExodusII file
+        """
```

Comment:
Doc strings quotations does not correspond the the one changes in other files. Or the method above.